spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -444,15 +444,20 @@ def main(argv: Sequence[str]):
 
     # Vertfile exists, so pre-process it if it's a `-discfile`
     else:
-        # Project discs labels to centerline for discfile
         if arguments.discfile is not None:
-            discs_projected = project_centerline(Image(fname_segmentation), Image(fname_vert_level))
+            # Copy the input files to the tempdir
             temp_folder = TempFolder(basename="process-segmentation")
-            path_tmp = temp_folder.get_path()
-            discs_projected.save(os.path.join(path_tmp, add_suffix(fname_vert_level, '_projected')))
-            ctl_projected = label_regions_from_reference(Image(fname_segmentation), discs_projected, centerline=True)
-            fname_vert_level = os.path.join(path_tmp, add_suffix(fname_vert_level, '_projected_centerline'))
-            ctl_projected.save(fname_vert_level)
+            path_tmp_seg = temp_folder.copy_from(fname_segmentation)
+            path_tmp_vert_level = temp_folder.copy_from(fname_vert_level)
+            # Project discs labels onto centerline
+            discs_projected = project_centerline(Image(path_tmp_seg), Image(path_tmp_vert_level))
+            discs_projected.save(add_suffix(path_tmp_vert_level, '_projected'), mutable=True)
+            # Use the projected disc labels to extract a labeled centerline from the input segmentation
+            ctl_projected = label_regions_from_reference(Image(path_tmp_seg), discs_projected, centerline=True)
+            ctl_projected.save(add_suffix(path_tmp_vert_level, '_projected_centerline'), mutable=True)
+            # Overwrite the input argument so that the labeled centerline (in the tmpdir) is used from now on
+            fname_vert_level = ctl_projected.absolutepath
+            # If requested, save the projected centerline to the output directory
             if verbose == 2:
                 copy(fname_vert_level, os.path.dirname(file_out))
```

Comment:
@valosekj has raised a point in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/5067#issuecomment-3471600690 suggesting it's not ideal that we're saving the `-discfile` to the same directory as the output `.csv` (presumably because the `.csv` may be saved to a "central location" for `sct_run_batch` scripts for `-append` purposes).

Since we're already modifying `-discfile` saving in this PR, I think it should be fine to also update this behavior here, too?